src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -123,10 +125,80 @@ private function doAsk(OutputInterface $output, Question $question)
             }
 
             if (false === $ret) {
-                $ret = fgets($inputStream, 4096);
-                if (false === $ret) {
-                    throw new MissingInputException('Aborted.');
+                if (!Terminal::hasSttyAvailable()) {
+                    $ret = fgets($inputStream, 4096);
+                    if (false === $ret) {
+                        throw new MissingInputException('Aborted.');
+                    }
+                } else {
+                    $sttyMode = shell_exec('stty -g');
+
+                    shell_exec('stty -icanon -echo');
+
+                    $k = null;
+                    $string = '';
+
+                    [$x] = $cursor->getCurrentPosition();
+
+                    while (10 !== $k && 0 !== $k) {
+                        [$pos] = $cursor->getCurrentPosition();
+                        $pos -= $x;
+
+                        $ret = fgetc($inputStream);
+                        if (false === $ret && '' === $string) {
+                            shell_exec(sprintf('stty %s', $sttyMode));
+                            throw new MissingInputException('Aborted.');
+                        }
+
+                        $k = \ord($ret);
```

Comment:
To make this code maintainable in the future, we could rename some variables. E.g. `$k` -> `$pressedKey` or `$keyCode`,etc.